documentation/src/main/asciidoc/reference/introduction.adoc

```diff
@@ -0,0 +1,517 @@
+driver for your database, for example:
+
+- `io.vertx:vertx-pg-client` for Postgres,
+- `io.vertx:vertx-mysql-client` for MySQL, or
+- `io.vertx:vertx-db2-client` for DB2.
+
+You don't need to depend on the JDBC driver for your database (but you
+can if you like).
+
+=== Optional dependencies
+
+Optionally, you might also add:
+
+- an {slf4j}[SLF4J] logging implementation, for example,
+  `org.slf4j:slf4j-log4j12`,
+- `org.hibernate:hibernate-jpamodelgen` if you're using the JPA criteria
+  query API,
+- `org.hibernate.validator:hibernate-validator` and
+  `org.glassfish:jakarta.el` if you're using Hibernate Validator,
+- `org.hibernate:query-validator` to enable compile-time checking of your
+  HQL queries, and/or
+- the Hibernate bytecode enhancer if you want to use field-level lazy
+  fetching.
+
+TIP: Field-level lazy fetching is an advanced feature that most programs
+don't need. Stick to the basics for now.
+
+There's an example {build}[Gradle build] included in the example program.
+
+=== Configuration
+
+:xml: https://github.com/hibernate/hibernate-reactive/blob/master/example/src/main/resources/META-INF/persistence.xml
+
+Hibernate Reactive is configured via the standard JPA `persistence.xml`
+document which must be placed, as usual, in the `/META-INF` directory.
+
+The only configuration specific to Hibernate Reactive is the persistence
+`<provider>` element, which must be explicit:
+
+[source,xml]
+----
+<provider>org.hibernate.reactive.provider.ReactivePersistenceProvider</provider>
+----
+
+Otherwise, configuration is almost completely transparent. Configure
+Hibernate exactly as you usually would, noting that most configuration
+properties related to JDBC or JTA aren't relevant in the context of
+Hibernate Reactive.
+
+Configuration properties of particular interest include:
+
+- `javax.persistence.jdbc.url`, the JDBC URL of your database,
+- `javax.persistence.jdbc.user` and `javax.persistence.jdbc.password`,
+the database credentials,
+- `javax.persistence.schema-generation.database.action`, for enabling
+  automatic schema export.
+
+An example {xml}[`persistence.xml`] file is included in the example
+program.
+
+=== Logging the generated SQL
+
+:log4j: https://github.com/hibernate/hibernate-reactive/blob/master/example/src/main/resources/log4j.properties
+
+To see the generated SQL as it's sent to the database, either:
+
+- set the property `hibernate.show_sql` to `true`, or
+- enable debug-level logging for the category `org.hibernate.SQL`
+  in for your preferred SLF4J logging implementation.
+
+For example, if you're using Log4J via `slf4j-log4j12` (as above in
+<<_optional_dependencies>>), add this line to your `log4j.properties`
+file:
+
+[source,properties]
+----
+log4j.logger.org.hibernate.SQL=DEBUG
+----
+
+An example {log4j}[`log4j.properties`] file is included in the example
+program.
+
+=== Tuning the Vert.x pool
+
+The Vert.x database client has built-in connection pooling and prepared
+statement caching.
```

Comment:
Caching is disabled for DB2 and MySQL

Reply:
Yes but that's a bug we're in the process of fixing. We' don't need to mention bugs in the user guide.